blocks/api/raw-handling/embedded-content-reducer.js

```diff
@@ -24,24 +24,16 @@ export default function( node ) {
 		return;
 	}
 
-	let nodeToInsert = node;
-	// if the embedded is an image and its parent is an anchor with just the image
-	// take the anchor out instead of just the image
-	if (
-		'IMG' === node.nodeName &&
-		1 === node.parentNode.childNodes.length &&
-		'A' === node.parentNode.nodeName
-	) {
-		nodeToInsert = node.parentNode;
-	}
+	// Consider parent anchor part of the embedded content.
+	const targetNode = node.parentElement.nodeName === 'A' ? node.parentElement : node;
```

Comment:
Shouldn't we just apply this to images, e.g: if we have <a><ifram...></a> I think we should not use this logic.
Also before there was verification to check if the anchor contains just one image. So in the multiple images case <a><img...><img></a> the images are individually taken out of the anchor (and then transformed in image block) right now we take out the anchor and all the images. Which might then end up not transforming all the images as before.